modules/circe/src/main/scala/ch/epfl/bluebrain/nexus/rdf/circe/JenaModel.scala

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.bluebrain.nexus.rdf.circe
+
+import ch.epfl.bluebrain.nexus.rdf.circe.JenaModel.JenaModelErr.{InvalidJsonLD, Unexpected}
+import ch.epfl.bluebrain.nexus.rdf.syntax.circe.context._
+import io.circe.Json
+import org.apache.jena.rdf.model.{Model, ModelFactory}
+import org.apache.jena.riot.system.StreamRDFLib
+import org.apache.jena.riot.{Lang, RDFParser, RiotException}
+
+import scala.util.Try
+
+object JenaModel {
+
+  /**
+    * @param json the json to be converted into a [[Model]]
+    * @return a [[Model]] if there were no errors an a [[JenaModelErr]] when we encounter errors
+    *         during model creation.
+    */
+  def apply(json: Json): Either[JenaModelErr, Model] = {
+    val base = json.hcursor.downField("@context").get[String]("@base").getOrElse("")
+    apply(json, base)
+  }
+
+  /**
+    *
+    * @param json the json to be converted into a [[Model]]
+    * @param base the JSON-LD base
+    * @return a Jena Model if there were no errors an a [[JenaModelErr]] when we encounter errors
+    *         during model creation.
+    */
+  def apply(json: Json, base: String): Either[JenaModelErr, Model] = {
```

Comment:
collapse the apply functions into a single one with `final def apply(json: Json, base: Option[AbsoluteIri])`